awx/api/serializers.py

```diff
@@ -906,6 +928,295 @@ def validate_is_superuser(self, value):
         return self._validate_ldap_managed_field(value, 'is_superuser')
 
 
+class UserAuthorizedTokenSerializer(BaseSerializer):
+    
+    refresh_token = serializers.SerializerMethodField()
+    token = serializers.SerializerMethodField()
+
+    class Meta:
+        model = OAuth2AccessToken
+        fields = (
+            '*', '-name', 'description', 'user', 'token', 'refresh_token',
+            'expires', 'scope', 'application',
+        )
+        read_only_fields = ('user', 'token', 'expires')
+        read_only_on_update_fields = ('application',)
+        
+    def get_token(self, obj):
+        request = self.context.get('request', None)
+        try:
+            if request.method == 'POST':
+                return obj.token
+            else:
+                return '*************'
+        except ObjectDoesNotExist:
+            return ''    
+        
+    def get_refresh_token(self, obj):
+        request = self.context.get('request', None)
+        try:
+            if request.method == 'POST':
+                return getattr(obj.refresh_token, 'token', '')
+            else:
+                return '**************'
+        except ObjectDoesNotExist:
+            return ''
+
+    def create(self, validated_data):
+        validated_data['user'] = self.context['request'].user
+        validated_data['token'] = generate_token()
+        validated_data['expires'] = now() + timedelta(
+            seconds=oauth2_settings.ACCESS_TOKEN_EXPIRE_SECONDS
+        )
+        obj = super(OAuth2TokenSerializer, self).create(validated_data)
+        obj.save()
+        if obj.application is not None:
+            OAuth2RefreshToken.objects.create(
+                user=self.context['request'].user,
+                token=generate_token(),
+                application=obj.application,
+                access_token=obj
+            )
+        return obj
+        
+
+class OAuth2ApplicationSerializer(BaseSerializer):
+
+    class Meta:
+        model = OAuth2Application
+        fields = (
+            '*', '-description', 'user', 'client_id', 'client_secret', 'client_type',
+            'redirect_uris',  'authorization_grant_type', 'skip_authorization',
+        )
+        read_only_fields = ('client_id', 'client_secret')
+        read_only_on_update_fields = ('user', 'authorization_grant_type')
+        extra_kwargs = {
+            'user': {'allow_null': False, 'required': True},
+            'authorization_grant_type': {'allow_null': False}
+        }        
+        
+    def to_representation(self, obj):
+        ret = super(OAuth2ApplicationSerializer, self).to_representation(obj)
+        if obj.client_type == 'public':
+            ret.pop('client_secret')
+        return ret
+        
+        
+    def get_modified(self, obj):
+        if obj is None:
+            return None
+        return obj.updated
```

Comment:
This method is defined in `BaseSerializer`, which is a parent class, and it performs the same functions as here. I see no reason to not delete this method.